TodoApp/TodoApp.Api.Tests/ItemsControllerTests.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+        [Test]
+        public async Task GetItemByIdAsync_ReturnsOkWithRequiredItem()
+        {
+            const string id = "1";
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.GetItemByIdAsync(id));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.Id, Is.EqualTo(id));
+            });
+        }
+
+        [Test]
+        public async Task PutItemAsync_ReturnsOkWithUpdatedItem()
+        {
+            const string id = "3";
+            var updatedItem = new Item
+            {
+                Id = id,
+                Text = "Write tests"
+            };
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.PutItemAsync(id, updatedItem));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.Id, Is.EqualTo(id));
+            });
+        }
+
+        [Test]
+        public async Task DeleteItemAsync_ReturnsOkWithDeletedItem()
+        {
+            const string id = "4";
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.DeleteItemAsync(id));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message.Content, Is.Not.Null);
+                Assert.That(message.TryGetContentValue<Item>(out var item), Is.Not.Null);
+                Assert.That(item.Id, Is.EqualTo(id));
+            });
+        }
+
+        [Test]
+        public async Task PostItemAsync_ReturnsCreatedWithLinkToItem()
+        {
+            _itemsController.Request = new HttpRequestMessage
+            {
+                RequestUri = new Uri("http://localhost/api/v1/items")
+            };
+            _itemsController.Configuration.Routes.MapHttpRoute(
+                name: "PostNewItem",
+                routeTemplate: "api/test/{id}",
+                defaults: new { id = RouteParameter.Optional }
+            );
+
+            var newItem = new Item
+            {
+                Id = "2",
+                Text = "Write tests"
+            };
+
+            var message = await ExecuteAsyncAction(async () => await _itemsController.PostItemAsync(newItem));
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(message.StatusCode, Is.EqualTo(HttpStatusCode.Created));
+                Assert.That(message.Headers.Location.AbsoluteUri, Is.EqualTo("http://localhost/api/test/2"));
```

Comment:
`http://localhost/api/test/2` move the id from the fake route inside the route. Just to make harder to fake test correctness.